### PR TITLE
Replace Read the Docs Sphinx Theme with Furo

### PR DESCRIPTION
**Pull request by hugovk**
Furo is a newer, more modern theme than [sphinx_rtd_theme](https://github.com/readthedocs/sphinx_rtd_theme), with native dark mode support. 

 * https://github.com/pradyunsg/furo
 * https://pradyunsg.me/furo/quickstart/

One thing I wasn't keen on before in Furo was the version selector floating over the main content near the bottom right, but it's now been moved into the left menu similar to sphinx_rtd_theme.

I think the code blocks and function headers look better in dark mode:

# Furo

image

image

# Old theme

image

image

# Demo

* https://pillow--6158.org.readthedocs.build/en/6158/

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -309,14 +306,6 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 # texinfo_no_detailmenu = False
 
-
-def setup(app):
-    app.add_js_file("js/script.js")
```
Comment (radarhere) [inline docs/conf.py:314]: This reverts #3476. Was that deliberate, or should I look at making script.js work with Furo?

File: docs/resources/css/dark.css [hunk 2]
```diff
@@ -1,1996 +0,0 @@
...
-        color: rgb(184, 178, 169);
-    }
-
-    .wy-body-for-nav .keep-us-sustainable a {
-        color: rgb(222, 219, 215);
-    }
-
-    /* For black-on-white/transparent images at handbook/text-anchors.html */
-    #text-anchors img {
-        filter: invert(1) brightness(0.85) hue-rotate(-60deg);
```
[the remark this answers is not in the document]
Comment (hugovk) [reply docs/resources/css/dark.css:1994]: Thanks! That didn't look so great:

image

Rebased back into place!

image

https://pillow--6158.org.readthedocs.build/en/6158/handbook/text-anchors.html#quick-reference

**Comment by hugovk** [comment]
Okay updated with this logic for dark mode + toggle combos!

light mode:

- 	light toggle: 	no invert
- 	auto toggle:	no invert
- 	dark toggle:	invert

dark mode:

- 	light toggle:	no invert
- 	auto toggle:	invert
- 	dark toggle:	invert

https://pillow--6158.org.readthedocs.build/en/6158/handbook/text-anchors.html#quick-reference

**Comment by nulano** [comment]
Looks like there is an error shown on the Image Formats page (but I can't see a warning message in the doccheck build log):
image

**Comment by hugovk** [comment]
Thanks removed. That was the only one. Strange it prints an error in the actual documentation, but doesn't issue a warning during build. Fee free to report at https://github.com/pradyunsg/furo :)

**Comment by nulano** [comment]
> Strange it prints an error in the actual documentation, but doesn't issue a warning during build.

Looks like it's a wontfix: https://github.com/pradyunsg/furo/discussions/125

**Comment by hugovk** [comment]
Ah well, thanks for checking!

File: docs/conf.py (hunk 1 above)
Comment (radarhere) [reply docs/conf.py:314]: Thanks for pointing out that feature.
Furo's way of doing things does mean that there isn't a neat way to implement #3476 anymore.

**Comment by hugovk** [comment]
If it's too soon to the release to test this properly, I'm happy for it to wait until afterwards so we can have more time.

**Comment by radarhere** [comment]
Yes, I think I've just run out of time to review this in detail before the release. But merging it after would also give us time to test it out over the next few months with documentation updates from other PRs.

**Comment by hugovk** [comment]
Sounds good 👍

**Pull request merged**